SetupDataPkg/ConfApp/ConfApp.c

```diff
@@ -414,7 +415,10 @@ ConfAppEntry (
         } else if ((KeyData.Key.UnicodeChar == 'y') ||
                    (KeyData.Key.UnicodeChar == 'Y'))
         {
-          ResetCold ();
+          // Prepare ResetData GUID
+          CopyGuid (&ResetData.ResetGuid, &gConfAppResetGuid);
+          gRT->ResetSystem (EfiResetCold, EFI_SUCCESS, sizeof (ResetData), &ResetData);
+          Print (L"reset done...\n");

```

Comment:
Reset does not always occur as soon as it is called, sometimes the reset takes a few seconds, and code can proceed to continue executing.

Adding a deadloop prevents the system from being able to execute anything else that would be unexpected. 

```suggestion
          Print (L"reset done...\n");
          CpuDeadloop();
```

Reply:
Suggestion take care